ui/plugins/pinboard.js

```diff
@@ -120,14 +120,39 @@ treeherder.controller('PinboardCtrl', [
 
         $scope.canSaveClassifications = function() {
             var thisClass = $scope.classification;
-            var canSave = $scope.hasPinnedJobs() && (thPinboard.hasRelatedBugs() ||
+            var canSave = $scope.hasPinnedJobs() && (thPinboard.hasRelatedBugs() && $scope.user.loggedin ||
                           thisClass.failure_classification_id !== 4 ||
                           $rootScope.currentRepo.repository_group.name === "try" ||
                           $rootScope.currentRepo.repository_group.name === "project repositories" ||
                           (thisClass.failure_classification_id === 4 && thisClass.text.length > 0));
             return canSave;
         };
 
+        // Dyanmic btn/anchor title for any combination of save conditions
+        $scope.saveClassificationTitle = function() {
+            var title = "";
+
+            if (!$scope.user.loggedin) {
+                title = title.concat("not logged in / ");
+            }
+            if (!$scope.canSaveClassifications()) {
+                title = title.concat("ineligible classification data / ");
+            }
+            if (!$scope.hasPinnedJobs()) {
+                title = title.concat("no pinned jobs");
+            }
```

Comment:
I also tried these conditions/strings as an array of key pairs to loop through, but it ended up taking a few lines more and seemed to have some angular refresh issues, so I stuck with simplicity.